### PR TITLE
React 16 + Ref Support

### PR DESCRIPTION
**Pull request by thomashuston**
I've been spending a bit of time this week seeing if we can get SeatGeek working with Percy for a project I'm working on. I'm pretty close, but was running into a few issues which I've addressed in this PR:

1. `@percy/react` was restricting peer dependencies to only React v15, but React v16 has been out for a year. I've updated the peer dependencies allow both v15 and v16, and updated the unit tests here to run against v16.

2. I discovered that `ref`s don't work properly because we were loading two copies of React on the page when snapshotting (it was included in both the `snapshots` and `render` bundles). I've updated the webpack config to extract `react` and `react-dom` into a `vendor` bundle to prevent it from loading multiple times, and added some integration tests here to verify that refs work. The tests failed before my changes, and now render properly with my updates.

### CODE REVIEW

File: integration-tests/react-percy/package.json [hunk 1]
```diff
@@ -6,9 +6,9 @@
     "percy": "react-percy"
   },
   "dependencies": {
-    "prop-types": "^15.5.10",
-    "react": "^15.6.1",
-    "react-dom": "^15.6.1"
+    "prop-types": "^15.6.2",
+    "react": "^16.5.2",
+    "react-dom": "^16.5.2"
```
Comment (thomashuston) [inline integration-tests/react-percy/package.json]: A while back we discussed adding better integration test coverage of multiple React versions. Might be worth a separate PR that adds coverage for both React 15 and 16 and both create-react-app 1 and 2.

File: package.json [hunk 2]
```diff
@@ -44,6 +45,9 @@
       "<rootDir>/integration-tests/",
       "<rootDir>/packages/"
     ],
+    "setupFiles": [
+      "raf/polyfill"
```
Comment (thomashuston) [inline package.json]: React 16 requires a `requestAnimationFrame` polyfill for tests to run properly.

File: packages/react-percy-snapshot-render/package.json [hunk 3]
```diff
@@ -15,16 +15,16 @@
     "lib"
   ],
   "peerDependencies": {
-    "react": "^15.4.0",
-    "react-dom": "^15.4.0"
+    "react": "^15.4.0 || ^16.0.0",
+    "react-dom": "^15.4.0 || ^16.0.0"
```
Comment (thomashuston) [inline packages/react-percy-snapshot-render/package.json]: This allows users to have either React 15 or React 16 in their projects without getting any peer dependency warnings.

File: packages/react-percy-ci/src/index.js [hunk 4]
```diff
@@ -25,9 +25,15 @@ export default async function run(percyConfig, percyToken) {
   const assets = await compile(percyConfig);
 
   const environment = new Environment();
+  const runtimeEntry = findEntryPath(assets, EntryNames.runtime);
```
Comment (thomashuston) [inline packages/react-percy-ci/src/index.js]: The runtime entry contains a manifest webpack uses to make sure imports work correctly across entries now that we're splitting some dependencies out into the vendor bundle. The vendor bundle contains `react` and `react-dom`, so that they're not included in multiple scripts.

File: packages/react-percy-snapshot-render/src/__tests__/render-tests.js [hunk 5]
```diff
@@ -17,6 +17,9 @@ jest.mock('redbox-react', () => ({ error }) =>
 );
 /* eslint-enable react/display-name, react/prop-types */
 
+// eslint-disable-next-line no-console
+console.error = jest.fn();
```
Comment (thomashuston) [inline packages/react-percy-snapshot-render/src/__tests__/render-tests.js]: This test purposefully causes an error to get thrown so we can verify it's getting rendered. React 16 has a new mechanism for catching errors and prints a console error here, but it's not really useful in the context of this test.

**Comment by thomashuston** [comment]
@timhaines any ideas why the `create-react-app` snapshots are rendering blank? I ran `react-percy --debug` on that integration test project locally and loaded it up, and everything looks fine. Are some scripts not getting uploaded properly or something?

File: package.json (hunk 2 above)
Comment (Robdel12) [reply package.json]: Looks like Jest `21.3.x` included the polyfill by default. Maybe we can bump the version of Jest here (or a separate PR?) https://github.com/facebook/jest/pull/4568#issuecomment-336562985
